p2p/discv5/routing_table_manager.py

```
@@ -47,21 +55,60 @@
     PongMessage,
 )
 from p2p.discv5.routing_table import (
-    FlatRoutingTable,
+    KademliaRoutingTable,
 )
 from p2p.discv5.typing import (
     NodeID,
 )
 
 
+@to_tuple
+def partition_enr_indices_by_size(enr_sizes: Sequence[int],
+                                  max_payload_size: int,
+                                  ) -> Generator[Tuple[int, ...], None, None]:
+    current_partition: Tuple[int, ...] = ()
+    current_partition_size = 0
+    for index, size in enumerate(enr_sizes):
+        if size > max_payload_size:
```

Seems like this should raise an error as some kind of invariant condition since it is probably not reasonable for us to handle the properly inside this helper function.  This implies that an extra check should be added at some higher level that filters out ENR records that are too large and logs something to signal something is wrong.